api/views.py

```diff
@@ -145,13 +147,17 @@ def _formatted_bad_request(format_):
 @api_view(['GET'])
 def enrichment_view(request):
     observable_name = request.query_params.get('query')
+    if not re.match(regex_domain, observable_name):
+        return Response({"message":"Invalid Observable"}, status=status.HTTP_400_BAD_REQUEST)
+    if not re.match(regex_ip, observable_name):
+        return Response({"message":"Invalid Observable"}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
the correct way is to use the serializers for these checks too.

The serializers checks the data in input and then provides the data to be returned

Please see how we have done it in [IntelOwl here](https://github.com/intelowlproject/IntelOwl/blob/master/api_app/api.py#L54)